src/Concerns/Extendable.php

```diff
@@ -35,6 +40,71 @@ public static function hasExtend(string $name): bool
         return array_key_exists($name, static::$extends);
     }
 
+    /**
+     * Register a pipe to be applied before an expectation is checked.
+     */
+    public static function pipe(string $name, Closure $handler): void
+    {
+        self::$pipes[$name][] = $handler;
+    }
+
+    /**
+     * Register an interceptor that should replace an existing expectation.
+     */
+    public static function intercept(string $name, string|Closure $filter, Closure $handler): void
+    {
+        if (is_string($filter)) {
+            $filter = fn ($value, ...$arguments): bool => $value instanceof $filter;
+        }
+
+        self::pipe($name, function ($next, ...$arguments) use ($handler, $filter): void {
+            /** @phpstan-ignore-next-line  */
+            if ($filter($this->value, ...$arguments)) {
```

Comment:
I know this is currently a single use trait, but we have no dependence on `$this->value` in this trait other than here, so if this was placed in a class that didn't have `$value` as a property, then wouldn't it blow up?

Reply:
May be cleaner to reverse this and do an early `return $next();` inside the `if` statement. Might have less visual overload.

Reply:
> I know this is currently a single use trait, but we have no dependence on $this->value in this trait other than here, so if this was placed in a class that didn't have $value as a property, then wouldn't it blow up?

Maybe I'm wrong, but here `$this` will refer to the binding that will be done when the pipe closure is executed (it is bound to Expectation on line 98 of this file), so it is not strictly tied to the trait 

don't know if I managed to explain myself, let me know if I'm not

Reply:
Of course, my bad 👍